src/ws/cockpitauth.c

```diff
@@ -550,8 +551,16 @@ create_creds_for_spawn_authenticated (CockpitAuth *self,
                              COCKPIT_CRED_CSRF_TOKEN, csrf_token,
                              NULL);
 
-  g_free (csrf_token);
+  if (!cockpit_json_get_object (results, "login-data", NULL, &login_data))
+    {
+      g_warning ("received bad login-data from %s", sl->command);
+      login_data = NULL;
+    }
 
+  if (login_data)
+    cockpit_creds_set_login_data (creds, login_data);
```

Comment:
The reason i did it this way is that if you call /login with a valid cookie that it has a session for it just responds straight away so if we want to include login-data in that response we need to keep it around.